cdap-common/src/main/java/io/cdap/cdap/common/metadata/QueryTerm.java

```diff
@@ -62,7 +62,12 @@ public Qualifier getQualifier() {
 
   @Override
   public boolean equals(Object o) {
-    return this.getClass() == o.getClass() && this.hashCode() == o.hashCode();
+    if (o == this) return true;
```

Comment:
This will not pass checkstyle, because you are missing the { .. }

Reply:
Addressed in latest commit.